File: tests/omp/target6.cpp
const int THRESHOLD1 = 1000000;

const int THRESHOLD2 = 1000;

void init(float*, float*, int);
void output(float*, int);

void vec_mult(float *p, float *v1, float *v2, int N)
{
   init(v1, v2, N);
   for (int i=0; i<N; i++)
     p[i] = v1[i] * v2[i];
   output(p, N);
}

File: tests/omp/target6.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->

  (* !! Omp.parallel_for ~clause:[If "N > THRESHOLD2"] [cFor_c ""]; *)
  
  !! Omp.target ~clause:[Parallel_c; For_c; If "target: N > THRESHOLD1"; If "parallel : N > THRESHOLD2";Map_c (To, ["v1[0:N]"; "v2[:N]"]); Map_c (From, ["p[0:N]"])] [cFor_c ""];

)
